lib/si-layer-cache/src/db/workspace_snapshot.rs

```diff
@@ -50,7 +50,7 @@ where
         let key = WorkspaceSnapshotAddress::new(&postcard_value);
         let cache_key: Arc<str> = key.to_string().into();
 
-        self.cache.insert(cache_key.clone(), value_clone).await;
+        self.cache.insert(cache_key.clone(), value_clone);
```

Comment:
It's interesting that the enqueued disk write is handled out-of-band of the `insert` method call, making it non-async. Kind of clever

Reply:
It's kind of what we did, too, with the persister, but we exposed more of that detail up front than they do. The disk cache is optional technically, so it makes sense that this is this way